### PR TITLE
tsconfig es2020 node

### PR DESCRIPTION
**Pull request by samejima-san**
Changing tsconfig files:  'target' to 'es2020'

### CODE REVIEW

**Comment by samejima-san** [comment]
> targetting wrong branch

I know, i have to switch it on the web version but publish from cli

File: packages/modules/error-handling-react/tsconfig.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "dom.iterable",
       "scripthost"
     ],
-    "module": "commonjs",
+    "module": "es2020",
```
Comment (kckst8) [inline packages/modules/error-handling-react/tsconfig.json:13]: let's undo this. We want to reset this package's config back to what it was previously (targeting 2017 i think? and commonjs)

File: tsconfig.json [hunk 2]
```diff
@@ -10,13 +10,13 @@
       "dom.iterable",
       "scripthost"
     ],
-    "module": "commonjs",
+    "module": "es2020",//commonjs
```
Comment (kckst8) [inline tsconfig.json:13]: should remove the commented code

File: packages/modules/web-viewer-react/tsconfig.json [hunk 3]
```diff
@@ -2,7 +2,9 @@
   "extends": "./node_modules/@itwin/build-tools/tsconfig-base.json",
   "compilerOptions": {
     "allowSyntheticDefaultImports": true,
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "moduleResolution": "node",
+    "module": "es2020"
```
Comment (aruniverse) [inline packages/modules/web-viewer-react/tsconfig.json:7]: you should not need to override module here.

take a look at:
- https://github.com/iTwin/itwinjs-core/blob/master/tools/build/tsconfig-base.json
- https://github.com/iTwin/viewer/blob/next/packages/modules/viewer-react/package.json#L21

we typically extend the tsconfig-base (which should always have module `commonjs`)
then in the package.json scripts, in the build:cjs we build using the tsconfig in our repo
and when we build:esm, we override and set the module to `es2020`

i think the main thing you need to do is override the `target` to `es2020` in the tsconfig.json of every package

**Comment by aruniverse** [comment]
Have you run the test-app (the one in packages/apps/web-viewer-test)?

I'm running into this locally:
image

**Comment by aruniverse** [comment]
closing this pr, too many conflicts, will just redo whats necessary for 4.0 release

**Pull request closed without merging**